cmd/machines.go

```diff
@@ -43,11 +44,101 @@ func init() {
 					setUserEnv, checkRequiredFlags, createMachine,
 				),
 			},
+			{
+				Name:  "list",
+				Usage: "List machines for an organization",
+				Flags: []cli.Flag{
+					orgFlag("org to the machine belongs to", true),
+					teamFlag("the team the machine belongs to", false),
+					destroyedFlag(),
+				},
+				Action: chain(
+					ensureDaemon, ensureSession, loadDirPrefs, loadPrefDefaults,
+					setUserEnv, checkRequiredFlags, listMachinesCmd,
+				),
+			},
 		},
 	}
 	Cmds = append(Cmds, machines)
 }
 
+func listMachinesCmd(ctx *cli.Context) error {
+	cfg, err := config.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	client := api.NewClient(cfg)
+	c := context.Background()
+
+	args := ctx.Args()
+	if len(args) > 0 {
+		return errs.NewUsageExitError("Too many arguments supplied.", ctx)
+	}
+
+	// Look up the target org
+	org, err := client.Orgs.GetByName(c, ctx.String("org"))
+	if err != nil {
+		return errs.NewExitError(envListFailed)
+	}
+	if org == nil {
+		return errs.NewExitError("Org not found.")
+	}
+	orgID := org.ID
+
+	state := primitive.MachineActiveState
+	if ctx.Bool("destroyed") {
+		state = primitive.MachineDestroyedState
+	}
+
+	teams, err := client.Teams.List(c, org.ID, ctx.String("team"), "")
+	if err != nil {
+		return errs.NewErrorExitError("Failed to retrieve metadata", err)
+	}
+	if len(teams) < 1 {
+		return errs.NewExitError("Machine teams not found")
+	}
+
+	var teamID *identity.ID
+	if ctx.String("team") != "" {
+		teamID = teams[0].ID
+	}
+
+	orgIDs := []*identity.ID{orgID}
+	machines, err := client.Machines.List(c, orgIDs, &state, teamID)
+	if err != nil {
+		return err
+	}
+
+	teamMap := make(map[identity.ID]primitive.Team, len(teams))
+	for _, t := range teams {
+		if t.Body.TeamType == primitive.MachineTeam {
+			teamMap[*t.ID] = *t.Body
+		}
+	}
+
+	fmt.Println("")
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 8, ' ', 0)
+	fmt.Fprintln(w, "ID\tNAME\tSTATE\tTEAM\tCREATION DATE")
+	fmt.Fprintln(w, " \t \t \t \t ")
+	for _, machine := range machines {
+		mID := machine.Machine.ID.String()
+		m := machine.Machine.Body
+		teamName := "-"
+		for _, m := range machine.Memberships {
+			team := teamMap[*m.Body.TeamID]
```

Comment:
we should be checking that the team was found (e.g. `team, ok := teamMap[*m.Body.TeamID]`) if it's not then error out, then we wouldn't need to check the team type again